### PR TITLE
fix: Allow resource extractor to reference multiple values

### LINKED ISSUE #132: bug: Resource Extractor can't handle multiple references to the same Resource
### What is the problem?

Currently, the resource extractor is using the same name for a resource to Value Share to the future stacks. For example, if a KMS Key is being extracted, and a stack wants to know the `keyId` and the `keyArn`, only the first exported value works. In this example, it would only export the `keyId`.

### Reproduction Steps

```typescript
    // Lambda function references two values from a KMS Key
    const key = new Key(stack, 'TestKey');
    new Function(stack, 'TestLambda', {
      code: Code.fromInline(`def handler(event, context)\n    print(event)`),
      handler: 'index.handler',
      runtime: Runtime.PYTHON_3_9,
      environment: {
        KEY_ARN: key.keyArn,
        KEY_ID: key.keyId,
      },
    });

    const synthedApp = app.synth();
    Aspects.of(app).add(
      new ResourceExtractor({
        extractDestinationStack: extractedStack,
        stackArtifacts: synthedApp.stacks,
        valueShareMethod: ResourceExtractorShareMethod.CFN_OUTPUT,
        resourceTypesToExtract,
      })
    );
    app.synth({ force: true });
```

### What did you expect to happen?

Both values are correctly referenced.

### What actually happened?

Both values reference the Key ID

### cdk-enterprise-iac version

latest

### Language

Typescript, Python

### Other information

_No response_

### PR DESCRIPTION
**Pull request by derrike**
Fixes #132 

This PR appends the attribute that is being looked up if the intrinsic function type is Fn::GetAtt. This will allow the resource extractor to create multiple exports if a user is attempting to reference multiple values in an object. For example, a `kmsId` and a `kmsArn`.

This is a breaking change, since it modifies the way the resources are exported. If a user is currently using the resource extractor and updates to a version with this fix, they may get unexpected results.

### CODE REVIEW

File: src/patches/resource-extractor/resourceExtractor.ts [hunk 1]
```diff
@@ -440,30 +440,39 @@ export class ResourceExtractor implements IAspect {
...
+    // Support for multiple references to the same object
+    if ('Fn::GetAtt' in intrinsic) {
+      const attribute = intrinsic['Fn::GetAtt'][1];
+      shareName = shareName + `:${attribute}`;
+    }
 
     if (this.valueShareMethod == ResourceExtractorShareMethod.SSM_PARAMETER) {
-      const paramName = `/${shareName.replace(':', '/')}`;
-      const paramLogicalId = `p${name}`;
+      shareName = shareName.replace(':', '/').replace(':', '/');
```
Comment (maafk) [inline src/patches/resource-extractor/resourceExtractor.ts:455]: do you need both `replace`s here?

**Review by maafk — CHANGES_REQUESTED** [review]
Looks great! Only one minor comment

File: src/patches/resource-extractor/resourceExtractor.ts (hunk 1 above)
Comment (derrike) [reply src/patches/resource-extractor/resourceExtractor.ts:455]: For some reason, yes - it doesn't work with just one. I was confused myself as to the why..

Comment (maafk) [reply src/patches/resource-extractor/resourceExtractor.ts:455]: can you try

```ts
shareName = shareName.replaceAll(':', '/')
```

Comment (derrike) [reply src/patches/resource-extractor/resourceExtractor.ts]: Thanks, fixed

**Review by maafk — APPROVED** [review]
Looks great! Thank you for your contribution!

**Pull request merged**